### DISCUSSION TITLE
Oracle form automation using WDIO

### DISCUSSION THREAD

**Discussion by MSL-GitHub** [webdriverio/webdriverio#12590] [Q&A]
Dear Team,
We use WDIO for every web automation project. My team is really happy with the tech. 
Unfortunately, the team is facing a challenge in our existing enterprise-level project. We are working on an Oracle EBS test automation project. 50% of the project is on the web, but another 50% is in Oracle forms (which are part of the same application, not a stand-alone part).
We know the WDIO interacts only with the web elements. We are searching for a solution for the Oracle forms automation in WDIO or any workaround using WDIO.
Our client is really happy with the WDIO automation; we don't want to change the automation stack for this (Oracle forms). 
I hope this community gives some valuable feedback for my query.

https://github.com/webdriverio/webdriverio/assets/158020320/c28a4c99-576f-46b4-8e9a-32d230de2a1a

**Comment by christian-bromann** [comment]
> We use WDIO for every web automation project. My team is really happy with the tech.

Thanks for the kind words!

> We are searching for a solution for the Oracle forms automation in WDIO or any workaround using WDIO.

I don't think there is any tool available that allows you to automate UI in Java applets 🤔 this is definitely not possible with WebdriverIO.

**Comment by troyyer** [comment]
@MSL-GitHub 
~this is a long shot but have a read of this medium article~ 
~https://medium.com/@Amr.sa/build-your-first-automation-test-for-a-desktop-app-using-wdio-eb9322f4ada3~

~The writer mentioned using webdriverio, appium and WinAppDriver to automate a desktop app Evernote.~ 
~Here is the GitHub repo of the same project~

~https://github.com/amrsa1/desktopAppAutomationTest~

~Hypothetically, you could use the multiremote feature~
~https://webdriver.io/docs/multiremote/~
~So webdriverio will drive the web capability and the appium and WinAppDriver would drive the Java applet?~
Sorry please ignore the above suggestion as WinAppDriver can't do java applets https://github.com/microsoft/WinAppDriver/issues/1876

I know it may make this comment off topic but you may want to also consider using one of the paid testing tools such as TestComplete or Tosca. In my company we use webdriver based frameworks such as Webdriverio for the web automation testing but for our old non-web systems we ended up going for a paid tool as they for some reason or the other just work on them.
